packages/tag/TagSelector.tsx

```diff
@@ -0,0 +1,88 @@
+import { h, Component, emit, prop } from 'skatejs';
+import styles from './Tag.scss';
+import { Tag } from './Tag';
+
+
+// public
+interface TagSelectorProps extends JSX.HTMLProps<HTMLElement | any> {
+  onTagChange?: Function,
+  delimiter?: string,
+  tags?: any
+}
+
+export class TagSelector extends Component<TagSelectorProps> {
+  static get is() { return 'bl-tag-selector' }
+  static get props() {
+    return {
+      tags: prop.array({
+        attribute: true
+      }),
+      delimiter: prop.string({
+        attribute: true
+      }),
+    }
+  }
+
+  tags = [];
+  delimiter = ' '; // default value is space ' '
+
+  private handleInput( event ) {
+    const lastChar = event.target.value.substr(-1);
+    const value = event.target.value.slice(0, -1).trim();
+    const isDelimiter = lastChar === this.delimiter;
+
+    if ( value && isDelimiter ) {
+      this.addTag(value);
+      event.target.value = '';
+    }
+  }
+
+  private addTag(value:string){
+    this.tags.push(value);
+    this.emitNewData( this.tags );
+  }
+
+  private removeTag(event: CustomEvent){
+    const index = this.tags.indexOf( event.detail.tag.innerHTML );
```

Comment:
hmm it isn't safe to rely on DOM, but we have the tag structured via projection, and are emitting whole element instance. So we should maybe redesign the API of that element first. Let's discuss this